### PR TITLE
Correctly unsnooze snoozed trips

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR fixes an issue where monitoring on snoozed trip did not resume automatically the next calendar day.

### CODE REVIEW

**Review by br648 — APPROVED** [review]
All good and no issues with time zones in the unit test.

**Comment by binh-dam-ibigroup** [comment]
> All good and no issues with time zones in the unit test.

Did it unsnooze a trip if you let it run? On mine it didn't, I'll investigate.

**Comment by binh-dam-ibigroup** [comment]
> > All good and no issues with time zones in the unit test.
> 
> Did it unsnooze a trip if you let it run? On mine it didn't, I'll investigate.

Never mind, the previous try might have been a flake.

**Comment by binh-dam-ibigroup** [comment]
As mentioned in the team meeting, it is the wrong check (if you snooze a monitored trip on a day the trip does not happen, it will not unsnooze), so I'll take it back.

**Comment by binh-dam-ibigroup** [comment]
@br648 and @JymDyerIBI Ready for review again!

File: src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripTest.java [hunk 1]
```diff
@@ -746,4 +746,65 @@ private void assertCheckMonitoredTrip(
...
+
+        MonitoredTrip modifiedTrip = Persistence.monitoredTrips.getById(monitoredTrip.id);
+        assertEquals(!shouldUnsnooze, modifiedTrip.snoozed);
+
+        // Clear the created trip.
+        PersistenceTestUtils.deleteMonitoredTrip(modifiedTrip);
+    }
+
+    private static Stream<Arguments> createCanUnsnoozeTripCases() {
+        // (Trips from above response above starts on Tuesday, June 9, 2020 at 8:40am and ends at 8:58am.)
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripTest.java]: Nit. "Trips from above response start on Tuesday".

**Review by br648 — APPROVED** [review]
Just one nit, but all good.

**Review by JymDyerIBI — APPROVED** [review]
Looks good!

**Pull request merged**